include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/EdgeFunctions.h

```diff
@@ -472,6 +482,149 @@ template <typename AnalysisDomainTy> class EdgeFunctions {
   getSummaryEdgeFunction(n_t Curr, d_t CurrNode, n_t Succ, d_t SuccNode) = 0;
 };
 
+// This class can be used as a singleton factory for EdgeFunctions that only
+// have constant data.
+//
+// EdgeFunction that have constant data only need to be allocated once and can
+// be turned into a singleton, which save large amounts of memory. The
+// automatic singleton creation can be enbaled for a EdgeFunction class by
+// inheriting from EdgeFunctionSingletonFactory and only allocating
+// EdgeFunction throught the provided createEdgeFunction method.
+//
+// Old unused EdgeFunction that were deallocated need to/should be clean from
+// the factory by either calling cleanExpiredEdgeFunctions or initializing the
+// automatic cleaner thread with initEdgeFunctionCleaner.
+template <typename EdgeFunctionType, typename CtorArgT>
```

Comment:
@pdschubert we should discuss template parameter naming to achieve consistency within phasar.

Reply:
EdgeFunctionType is fine here. We consistently use that type name for the flow/edge function memory manager, too.